### DISCUSSION TITLE
Theming and styling components

### DISCUSSION THREAD

**Discussion by jkavalec** [vanjs-org/van#332] [General]
Hi,
after doing my annual search for 'JS framework without JSX' and this time unexpectedly getting result I've found this very nice project.

I like most of the things (docs are especially nice), and only thing I didn't find was some reasonable approach for Styling components as everything uses just prop object for styles and especially Themes that can be easily switched. I can imagine reasonably simple solution for components, and some more complex for theming, but wanted to check out whether this is something that has reasonable solution within VanJS already done.

Jarek

**Comment by Tao-VanJS** [comment]
For styling, you can refer to the simple solution used in VanUI: https://github.com/vanjs-org/van/blob/main/components/van-ui.ts.
